### ISSUE TITLE
Generating PropertiesChanged signals

### ISSUE THREAD

**Issue by agrover** [stratis-storage/stratisd#201]
While it's not needed for stratis-cli, the DBus API must generate PropertiesChanged signals. We need to investigate dbus-rs's support for generating these -- whether it's some automatic thing the lib does on our behalf, or if not then what we need to do to generate them.

**Comment by agrover** [comment]
From a brief look at the dbus-rs code, it appears that a property can automatically issue the signal when changed if set, by using the `emits_changed` builder method, but based upon advice we are avoiding writable properties, so this isn't helpful. We will likely need to generate signals ourselves when the value changes, either from being set by a `SetFoo` method or some other more indirect thing that causes the property value to change.

**Comment by mulkieran** [comment]
No longer part of this milestone, still part of the final plan.

**Comment by mulkieran** [comment]
We can take another look at this: https://github.com/diwic/dbus-rs/issues/66#issuecomment-328357598.

**Comment by agrover** [comment]
depends on #838

**Comment by agrover** [comment]
fixed by #1125
